File: rust/src/fizzbuzz.rs
fn get_fizzbuzz_equivalent(number: usize) -> String {
    let mut result = String::new();
    if number % 3 == 0 {
        result.push_str("Fizz");
    }
    if number % 5 == 0 {
        result.push_str("Buzz");
    }
    if result.is_empty() {
        return number.to_string();
    } else {
        return result;
    };
}

#[cfg(test)]
mod tests {
    use super::get_fizzbuzz_equivalent;
    #[test]
    fn should_return_the_number_in_string() {
        assert_eq!(get_fizzbuzz_equivalent(1), "1");
        assert_eq!(get_fizzbuzz_equivalent(124), "124");
    }

    #[test]
    fn should_return_fizz_if_multiple_of_3() {
        assert_eq!(get_fizzbuzz_equivalent(3), "Fizz");
        assert_eq!(get_fizzbuzz_equivalent(93), "Fizz");
    }

    #[test]
    fn should_return_buzz_if_multiple_of_5() {
        assert_eq!(get_fizzbuzz_equivalent(5), "Buzz");
        assert_eq!(get_fizzbuzz_equivalent(95), "Buzz");
    }

    #[test]
    fn should_return_fizzbuzz_if_multiple_of_3_and_5() {
        assert_eq!(get_fizzbuzz_equivalent(15), "FizzBuzz");
        assert_eq!(get_fizzbuzz_equivalent(90), "FizzBuzz");
    }
}

File: zig/src/fizzbuzz.zig
const std = @import("std");
var gpa = std.heap.GeneralPurposeAllocator(.{}){};
const allocator = gpa.allocator();

pub fn get_fizzbuzz(comptime n: usize) []const u8 {
    if (n % 15 == 0) {
        return "FizzBuzz";
    } else if (n % 3 == 0) {
        return "Fizz";
    } else if (n % 5 == 0) {
        return "Buzz";
    }
    return std.fmt.comptimePrint("{}", .{n});
}

const testing = std.testing;

test "Should return the passed number in a string" {
    try testing.expectEqualSlices(u8, get_fizzbuzz(1), "1");
    try testing.expectEqualSlices(u8, get_fizzbuzz(17), "17");
}

test "Should return Fizz is the number is multiple of 3" {
    try testing.expectEqualSlices(u8, get_fizzbuzz(3), "Fizz");
    try testing.expectEqualSlices(u8, get_fizzbuzz(3 * 34), "Fizz");
}

test "Should return Buzz is the number is multiple of 5" {
    try testing.expectEqualSlices(u8, get_fizzbuzz(5), "Buzz");
    try testing.expectEqualSlices(u8, get_fizzbuzz(5 * 34), "Buzz");
}

test "Should return FizzBuzz is the number is multiple of 3 and 5" {
    try testing.expectEqualSlices(u8, get_fizzbuzz(3 * 5), "FizzBuzz");
    try testing.expectEqualSlices(u8, get_fizzbuzz(3 * 5 * 34), "FizzBuzz");
}